lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1820,8 +1820,17 @@ TypeSystemSwiftTypeRef::RemangleAsType(swift::Demangle::Demangler &dem,
 swift::Demangle::NodePointer TypeSystemSwiftTypeRef::DemangleCanonicalType(
     swift::Demangle::Demangler &dem, opaque_compiler_type_t opaque_type) {
   using namespace swift::Demangle;
-  CompilerType type = GetCanonicalType(opaque_type);
-  return GetDemangledType(dem, type.GetMangledTypeName().GetStringRef());
+  NodePointer canonical =
+      GetCanonicalDemangleTree(dem, AsMangledName(opaque_type));
+  if (ContainsUnresolvedTypeAlias(canonical)) {
+    // If this is a typealias defined in the expression evaluator,
+    // then we don't have debug info to resolve it from.
+    CompilerType ast_type =
+        ReconstructType({this, opaque_type}).GetCanonicalType();
+    canonical = GetCanonicalDemangleTree(
+        dem, ast_type.GetMangledTypeName().GetStringRef());
```

Comment:
this logic was inlined (copied) from `GetCanonicalType` in the same file.

Reply:
Should it be a static helper maybe?